### PR TITLE
Material EditText

### PR DESCRIPTION
**Pull request by rastislavvasko**
This adds a `MaterialEditText` with a tinted background and cursor.

Also added is partial support for the `android:theme` attribute for widgets. Partial, because the themed context won't be passed to the widget's descendents.

Lastly, `MaterialWidgetHandler` now supports initializing drawables by raw color value.

### CODE REVIEW

File: MaterialLibrary/src/main/java/io/doist/material/widget/RobotoEditText.java [hunk 1]
```diff
@@ -3,11 +3,10 @@
 import android.content.Context;
 import android.graphics.Typeface;
 import android.util.AttributeSet;
-import android.widget.EditText;
 
 import io.doist.material.utils.Roboto;
 
-public class RobotoEditText extends EditText {
+public class RobotoEditText extends MaterialEditText {
```
Comment (jpmcosta) [inline MaterialLibrary/src/main/java/io/doist/material/widget/RobotoEditText.java]: I think that `MaterialEditText` is the one that should extend from `RobotoEditText`.

File: MaterialLibrary/src/main/java/io/doist/material/widget/utils/MaterialWidgetHandler.java [hunk 2]
```diff
@@ -1,27 +1,34 @@
...
 
+import java.lang.reflect.Array;
+
+import io.doist.material.R;
 import io.doist.material.reflection.ReflectionUtils;
 import io.doist.material.res.MaterialResources;
 
 public class MaterialWidgetHandler {
-    private static final boolean sSkip = Build.VERSION.SDK_INT > Build.VERSION_CODES.KITKAT;
+    private static final boolean SKIP = Build.VERSION.SDK_INT > Build.VERSION_CODES.KITKAT;
```
Comment (jpmcosta) [inline MaterialLibrary/src/main/java/io/doist/material/widget/utils/MaterialWidgetHandler.java]: :+1:

File: MaterialLibrary/src/main/java/io/doist/material/widget/utils/MaterialWidgetHandler.java [hunk 3]
```diff
@@ -151,4 +195,40 @@ public static void init(View view, AttributeSet set, int defStyle, int[] attrs)
         }
         return newArray;
     }
+
+    private static Drawable getDrawable(MaterialResources r, TypedArray ta, int index) {
```
Comment (jpmcosta) [inline MaterialLibrary/src/main/java/io/doist/material/widget/utils/MaterialWidgetHandler.java]: I don't think `MaterialWidgetHandler` should handle any part of `getDrawable`. We should always use `MaterialResources`. More so after the merge with https://github.com/Doist/AndroidMaterial/commit/eafcefa0694ea20aafb9ababfcf9bf075c113bd3.
What do you think?

**Comment by jpmcosta** [comment]
This seems good.

Some notes:
- I believe that we won't be needing `MaterialContextThemeWrapper` after the merge with `ft.material_dialog`, since we will use reflection to get the theme from a `ContextThemeWrapper`;
- I think that `ft.material_dialog` version of `MaterialResources` can safely replace this branch's version. However, it would be very nice if you could review it, to make sure it is working and whether or not it covers all the functionality needed here.

We should obviously discuss any matter you find relevant, because my comments are based on my idea on how we can merge this branch with `ft.material_dialog`.

**Comment by rastislavvasko** [comment]
This is now based on `ft.material_dialog` (after it was merged into `master`).

In fd9ff8da684824f1d6117362cdee65a031a4f276 the previous changes to `MaterialResources` and `MaterialWidgetHandler` were completely replaced with the version in `master` and later commits only restore the necessary functionality to make reviewing easier.

File: MaterialLibrary/src/main/java/io/doist/material/widget/utils/MaterialWidgetHandler.java [hunk 4]
```diff
@@ -256,4 +350,21 @@ private static void initTextViewAttributes(Context context, MaterialResources re
         }
         return newStyleable;
     }
+
+    private static int getThemeResId(Context context) {
```
Comment (jpmcosta) [inline MaterialLibrary/src/main/java/io/doist/material/widget/utils/MaterialWidgetHandler.java]: We can probably turn `MaterialResources$getThemeResId(Context)` into a public method. What do you think?

**Comment by jpmcosta** [comment]
The PR seems good.
Some notes:
- I think that we should add `themifyContext` to all material widgets, like `MaterialButton`, `MaterialEditText` and `MaterialView`.
- I also believe we should standardize our calls to `super()` or `this()` in the `Material<Widget>` constructors. I think we should use `this()` whenever possible.
- I think we could also fit a small change to `RobotoEditText` and `RobotoTextView` – when calling `applyRoboto` the default `defStyle` should be `android.R.attr.editTextStyle` and `android.R.attr.textViewStyle`, respectively. Currently we are just passing `0`.

**Comment by rastislavvasko** [comment]
Thanks for review! Fully agree with feedback and everything has been implemented. :)

**Comment by jpmcosta** [comment]
Everything good! :+1:

**Pull request merged**